packages/protocol/contracts/authorizers/RankedAdminnable.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "./interfaces/IRankedAdminnable.sol";
+
+/// @title Contract that implements multiple ranks of admins
+contract RankedAdminnable is IRankedAdminnable {
+  /// @notice Keeps the ranks of admins for each individual adminned entity
+  /// @dev This contract implements adminship for a mapping of entities, rather
+  /// than a single entity. However, the logic at the inheriting contract can
+  /// easily be adapted to use this contract to admin a single entity (see
+  /// Api3CwRrpAuthorizer.sol)
+  mapping(bytes32 => mapping(address => uint256)) public adminnedIdToAdminToRank;
```

Comment:
I'm wondering if we really need uint256 for holding ranks 🤔 it seems to me that even with an uint8 or uint16 might have enough ranks for an entity (I might be missing something here). Also setting MAX_RANK in this contract seems to make more sense to me.

Reply:
Yeah 256 bits is overkill, but using a smaller type doesn't save any gas (when you read a single rank one full word will be read from the storage, when you read a single rank, one full word will be written to the storage). Future forks may easily change this but I don't find it a good idea to try to predict non-existent EIPs and using a smaller type than 256 bits is additional complexity so I'd rather avoid it unless there is no advantage.
I also thought about including `MAX_RANK` in this contract. Obviously the reason I didn't is that it's not used in it but then it looks like all the inheriting contracts will need to use it in some form so maybe it should be added. Again, let me think and let's get @Ashar2shahid 's opinion.

Reply:
Yeah there probably should be `MAX_RANK`, also there might be a need for a modifier that is onlyMaxRank, because I can see it being reused alot.